frontend/packages/ceph-storage-plugin/src/components/dashboards/ocs-dashboards.tsx

```diff
@@ -0,0 +1,69 @@
+import * as React from 'react';
+import { connect } from 'react-redux';
+import { Helmet } from 'react-helmet';
+import { useTranslation } from 'react-i18next';
+import { useFlag } from '@console/shared/src/hooks/flag';
+import { useExtensions, DashboardsCard, isDashboardsCard } from '@console/plugin-sdk';
+import Dashboard from '@console/shared/src/components/dashboard/Dashboard';
+import DashboardGrid, {
+  GridPosition,
+} from '@console/shared/src/components/dashboard/DashboardGrid';
+import {
+  getCardsOnPosition,
+  mapStateToProps,
+  DashboardsPageProps,
+} from '@console/internal/components/dashboard/dashboards-page/dashboards';
+import { HorizontalNav, PageHeading, LoadingBox } from '@console/internal/components/utils';
+import { OCS_INDEPENDENT_FLAG } from '../../features';
+
+export const getTabCards = (tabId: string, cards: DashboardsCard[]) => {
+  const tabCards = cards.filter((c) => c.properties.tab === tabId);
+  return () => (
+    <Dashboard>
+      <DashboardGrid
+        mainCards={getCardsOnPosition(tabCards, GridPosition.MAIN)}
+        leftCards={getCardsOnPosition(tabCards, GridPosition.LEFT)}
+        rightCards={getCardsOnPosition(tabCards, GridPosition.RIGHT)}
+      />
+    </Dashboard>
+  );
+};
+
+const OCSDashboardsPage: React.FC<DashboardsPageProps> = ({ match, kindsInFlight, k8sModels }) => {
+  const { t } = useTranslation();
+  const title = t('ceph-storage-plugin~OpenShift Container Storage Overview');
+  const isOcsIndependent = useFlag(OCS_INDEPENDENT_FLAG);
+  const cardExtensions = useExtensions<DashboardsCard>(isDashboardsCard);
```

Comment:
You're still using extensions meant for cluster overview. You need to create your own.

Reply:
or you could use the same but you need a way to identify to which dashboard the extension should be added. Maybe some URL matching

Reply:
```suggestion
  const cardExtensions = useExtensions<DashboardsCard>((e) => isDashboardsCard && (['independent-dashboard', 'persistent-storage', 'object-service'].includes(e?.properties?.tab);
```
We should have unique tab identifiers and this would do the trick. @rawagner wdyt?
I guess getTabCards is also doing something similar so do we really need a new extension ?

Reply:
We are adding new OCS tabs without using the extension (in `ocs-dashboards.tsx` file; Not in `plugin.ts` file). These OCS cards will not overlap/interfere with any of the cluster overview tabs.

Or, if that is not the right way, we could add one more property to the extension type: 'Dashboards/Tab', let's say navSection which specifies the nav section to which that dashboard tab belongs to. For example: for cluster overview tabs value will be “cluster” and for OCS specific tabs it will be “storage” and we will add one more filter on this property while displaying the dashboards.
So, the extension will look something like this:
```
{
   type: 'Dashboards/Tab',
   properties: {
     id: 'persistent-storage',
     navSection: 'storage',
     // t('ceph-storage-plugin~Persistent Storage')
     title: '%ceph-storage-plugin~Persistent Storage%',
   },
   flags: {
     required: [FLAG],
   },
 }
```
What do you guys think should be better?

Reply:
+1 for adding `navSection`